Eigen/Core

```diff
@@ -270,6 +301,19 @@
   #include <cuda_fp16.h>
 #endif
 
+#if defined(EIGEN_HIPCC) && defined(EIGEN_HIP_DEVICE_COMPILE)
+  #define EIGEN_HAS_HIP_FP16
+#endif
+
+#if defined EIGEN_HAS_HIP_FP16
```

Comment:
does the EIGEN_HAS_HIP_FP16 macro get used or defined anywhere else? If possible, maybe merge this block with the previous where EIGEN_HAS_HIP_FP16 gets defined.

Reply:
Nope...this is the only place it gets defined. The usage pattern simply mirrors what is in place for EIGEN_HAS_CUDA_FP16.  I had thought merging the two just like you mention, but decided to keep it separate. No particular reason for the choice.